src/main/java/org/apache/ibatis/scripting/xmltags/ForEachSqlNode.java

```diff
@@ -154,7 +154,7 @@ public void appendSql(String sql) {
       GenericTokenParser parser = new GenericTokenParser("#{", "}", new TokenHandler() {
         @Override
         public String handleToken(String content) {
-          String newContent = content.replaceFirst("^\\s*" + item + "(?![^.,:\\s])", itemizeItem(item, index));
+          String newContent = content.replaceFirst("^\\s*" + item + "(?![^.\\[,:\\s])", itemizeItem(item, index));
           if (itemIndex != null && newContent.equals(content)) {
             newContent = content.replaceFirst("^\\s*" + itemIndex + "(?![^.,:\\s])", itemizeItem(itemIndex, index));
```

Comment:
When a map is passed to `collection`, `itemIndex` represents the key of each entry.
So, technically, it can be a list or array, I think.
(I haven't had time to verify it)